packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
```

Comment:
No. If there is any setter in the prototype chain, it will be executed and will never goto `else` branch.